sui-execution/latest/sui-adapter/src/adapter.rs

```diff
@@ -83,7 +86,19 @@ mod checked {
     pub fn new_move_vm(
         natives: NativeFunctionTable,
         protocol_config: &ProtocolConfig,
+        _enable_profiler: Option<PathBuf>,
     ) -> Result<MoveVM, SuiError> {
+        #[cfg(feature = "gas-profiler")]
+        let vm_profiler_config = _enable_profiler.map(|_| VMProfilerConfig {
+            full_path: _enable_profiler.filter(|p| {
+                !matches!(
+                    p.partial_cmp(&*DEFAULT_PROFILE_OUTPUT_PATH),
+                    Some(std::cmp::Ordering::Equal)
+                )
```

Comment:
Not that it should matter based on my comment above, but isn't this just `p == &*DEFAULT_PROFILER_OUTPUT_PATH`? Which is also problematic because it's not going to detect cases where someone passes in a path like `foo/..` which is actually the current working directory, but given in a non-canonical form (anyway, for me, that's just another vote for not trying to detect this at this stage).